### PR TITLE
FIX #4419 : added margin support in text widget

### LINKED ISSUE #4419: [Feature] Text widget should have padding controls
## Summary
The text widget has no padding controls so it looks ugly when you color the background.

Screen Shot 2021-05-11 at 4 32 32 AM
Providing padding controls will allow the developer to create better looking text cards.

### PR DESCRIPTION
**Pull request by vibhandikyash**
## Description

> Added margin support in text widget.

Fixes #4419 

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

> Tested Locally.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: FEATURE/4419-text-widget-padding-controls 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 53.84 **(0)** | 35.81 **(0)** | 32.43 **(-0.01)** | 54.42 **(0)**
 :red_circle: | app/client/src/components/designSystems/blueprint/TextComponent.tsx | 41.18 **(-2.57)** | 10 **(0)** | 8.33 **(-0.76)** | 39.39 **(-2.55)**</details>

### CODE REVIEW

File: app/client/src/widgets/TextWidget.tsx [hunk 1]
```diff
@@ -139,6 +139,16 @@ class TextWidget extends BaseWidget<TextWidgetProps, WidgetState> {
             isBindProperty: false,
             isTriggerProperty: false,
           },
+          {
+            helpText: "Enter value for margin",
```
Comment (somangshu) [inline app/client/src/widgets/TextWidget.tsx:143]: Margin allows clean space between text and widget boundary

File: app/client/src/components/designSystems/blueprint/TextComponent.tsx [hunk 2]
```diff
@@ -30,11 +30,13 @@ export const StyledText = styled(Text)<{
   textColor?: string;
   fontStyle?: string;
   fontSize?: TextSize;
+  margin?: string;
```
Comment (somangshu) [inline app/client/src/components/designSystems/blueprint/TextComponent.tsx:33]: This dosent work, I was wrong here, `Padding` will work here. Please check the image in original issue to get clarity.

File: app/client/src/widgets/TextWidget.tsx [hunk 3]
```diff
@@ -139,6 +139,16 @@ class TextWidget extends BaseWidget<TextWidgetProps, WidgetState> {
...
+            propertyName: "margin",
+            label: "Margin",
```
Comment (somangshu) [inline app/client/src/widgets/TextWidget.tsx:145]: This should finally be `padding (px)` and the value should be an integer which works without px

**Review by somangshu — CHANGES_REQUESTED** [review]

**Review by somangshu — APPROVED** [review]

File: app/client/src/widgets/TextWidget.tsx [hunk 4]
```diff
@@ -140,14 +140,14 @@ class TextWidget extends BaseWidget<TextWidgetProps, WidgetState> {
             isTriggerProperty: false,
           },
           {
-            helpText: "Enter value for margin",
-            propertyName: "margin",
-            label: "Margin",
-            placeholderText: "Enter value in px",
+            helpText: "Enter value for padding (px)",
```
Comment (somangshu) [inline app/client/src/widgets/TextWidget.tsx:143]: Change to > "Controls clean space between text and widget boundary"

**Review by somangshu — APPROVED** [review]

File: app/client/src/widgets/TextWidget.tsx [hunk 5]
```diff
@@ -139,6 +139,16 @@ class TextWidget extends BaseWidget<TextWidgetProps, WidgetState> {
             isBindProperty: false,
             isTriggerProperty: false,
           },
+          {
+            helpText: "Controls clean space between text and widget boundary",
+            propertyName: "padding",
+            label: "Padding",
```
Comment (somangshu) [inline app/client/src/widgets/TextWidget.tsx:145]: change to > "Padding (px)"

**Review by somangshu — CHANGES_REQUESTED** [review]
Change order of `Padding` > above `font style`

**Review by somangshu — APPROVED** [review]

**Comment by vivekverma2312** [comment]
@vibhandikyash If I use bigger padding then the text is displayed out of the widget due to this it overlaps the below widget.
Screenshot 2021-07-15 at 2 19 33 PM

**Comment by somangshu** [comment]
@marks0351 any solutions around this?

**Comment by marks0351** [comment]
> @marks0351 any solutions around this?

we should be decreasing the height of widget based on the padding added. widget content should not be overflowing its borders it can overflow within it selves like we do with containers.

**Comment by marks0351** [comment]
@somangshu should we temporarily close this PR?

**Comment by somangshu** [comment]
@vibhandikyash please have a discussion with @marks0351 and take this forward.

**Comment by vibhandikyash** [comment]
@marks0351 As we discussed. This is dependent on #2587. Once that is merged. We can take this further.
cc: @somangshu

**Comment by somangshu** [comment]
This will be picked up after the container widget branch is merged, Closing this for now

**Pull request closed without merging**